pkg/controller/persistentvolume/persistentvolume_controller.go

```diff
@@ -236,3 +256,11 @@ func (ctrl *PersistentVolumeController) Stop() {
 		delete(ctrl.stopChannels, name)
 	}
 }
+
+// isFullySynced returns true, if both volume and claim caches are fully loaded
+// after startup.
+// We do not want to process events with not fully loaded caches - e.g. we might
+// recycle/delete PVs that don't have corresponding claim in the cache yet.
+func (ctrl *PersistentVolumeController) isFullySynced() bool {
+	return ctrl.volumeController.HasSynced() && ctrl.claimController.HasSynced()
```

Comment:
Are we guaranteed that once this condition is true, that we will call Sync*() immediately?  Where is that guarantee satisfied?